src/ConsentRequiredController.js

```diff
@@ -0,0 +1,135 @@
+define([
+  'okta',
+  'shared/util/Util',
+  'util/FormController',
+  'util/FormType',
+  'views/consent/ConsentHeader',
+  'views/consent/ConsentBeacon',
+  'views/consent/ScopeList',
+  'views/consent/ConsentButtonsBar'
+],
+function (Okta, Util, FormController, FormType, ConsentHeader, ConsentBeacon, ScopeList, ConsentButtonsBar) {
+
+  var $ = Okta.$,
+      _ = Okta._;
+
+  return FormController.extend({
+    className: 'consent-required',
+    initialize: function () {
+      this.model.set('expiresAt', this.options.appState.get('transaction').expiresAt);
+      this.model.set('scopes', this.options.appState.get('transaction').scopes);
+    },
+    Model: {
+      props: {
+        expiresAt: ['string', true],
+        scopes: ['array', true]
+      },
+      save: function () {
+        return this.doTransaction(function(transaction) {
+          return transaction.consent({
+            consent: {
+              expiresAt: this.get('expiresAt'),
+              scopes: _.pluck(this.get('scopes'), 'name')
+            }
+          });
+        });
+      },
+      cancel: function () {
+        var self = this;
+        return this.doTransaction(function(transaction) {
+          return transaction.cancel();
+        }).then(function () {
+          Util.redirect(self.settings.get('consentCancelLink'));
+        });
+      }
+    },
+    Form: {
+      noButtonBar: true,
+      formChildren: function () {
+        var transaction = this.options.appState.get('transaction');
+        return [
+          FormType.View({
+            View: new ConsentHeader({
+              orgLogo: this.settings.get('logo'),
+              userFirstName: transaction.user.profile.firstName,
+              userLastName: transaction.user.profile.lastName
+            })
+          }),
+          FormType.View({
+            View: new ConsentBeacon({
+              clientLogo: transaction.target.logo && transaction.target.logo.href,
+            })
+          }),
+          FormType.View({
+            View: Okta.View.extend({
+              className: 'consent-title',
+              template: '\
+                <p><b>{{appName}}</b> {{i18n code="consent.required.title" bundle="login"}}</p>\
+              ',
+              getTemplateData: function () {
+                return { appName: transaction.target.label };
+              }
+            })
+          }),
+          FormType.View({
+            View: new ScopeList({ model: this.model })
+          }),
+          FormType.View({
+            View: Okta.View.extend({
+              className: 'consent-description',
+              template: '\
+                <p>{{i18n code="consent.required.description" bundle="login"}}</p>\
+                {{#if termsOfService}}\
+                  <a class="terms-of-service" href="{{termsOfService}}" target="_blank">\
+                    {{i18n code="consent.required.termsOfService" bundle="login"}}\
+                  </a>\
+                  {{#if privacyPolicy}}\
```

Comment:
this will not work if we only have ```privacyPolicy``` since it will show a "separator" before 'Privacy Policy' with nothing before it

Reply:
no is nested inside termsOfService

Reply:
ah right, missed that. 👍